backend/src/test/java/com/bakdata/conquery/api/form/config/FormConfigTest.java

```diff
@@ -1,441 +1,341 @@
+	private AuthorizationController controller;
+	private Validator validator = Validators.newValidatorFactory().getValidator();
+	
+	private Dataset dataset = new Dataset();
+	private Dataset dataset1 = new Dataset();
+	private DatasetId datasetId;
+	private DatasetId datasetId1;
+	private ExportForm form;
+	
+	@BeforeAll
+	public void setupTestClass() throws Exception{
+
+		dataset.setName("test");
+		dataset1.setName("test1");
+		datasetId = dataset.getId();
+		datasetId1 = dataset1.getId();
+
+		// Mock DatasetRegistry for translation
+		namespacesMock = Mockito.mock(DatasetRegistry.class);
+		doAnswer(invocation -> {
+			throw new UnsupportedOperationException("Not yet implemented");
+		}).when(namespacesMock).getOptional(any());
+		doAnswer(invocation -> {
+			final DatasetId id = invocation.getArgument(0);
+			Namespace namespaceMock = Mockito.mock(Namespace.class);
+			if(id.equals(datasetId)) {
+				when(namespaceMock.getDataset()).thenReturn(dataset);
+			}
+			else if (id.equals(datasetId1)) {
+				when(namespaceMock.getDataset()).thenReturn(dataset1);
+			}
+			else {
+				throw new IllegalStateException("Unkown dataset id.");
+			}
+			return namespaceMock;
+		}).when(namespacesMock).get(any(DatasetId.class));
+		when(namespacesMock.getAllDatasets()).thenReturn(List.of(dataset,dataset1));
+		when(namespacesMock.injectInto(any(ObjectMapper.class))).thenCallRealMethod();
+		when(namespacesMock.inject(any(MutableInjectableValues.class))).thenCallRealMethod();
+
+		storage = new MetaStorage(null, new NonPersistentStoreFactory(), Collections.emptyList(), namespacesMock);
+
+
+		((MutableInjectableValues)FormConfigProcessor.getMAPPER().getInjectableValues())
+		.add(IdResolveContext.class, namespacesMock);
+		processor = new FormConfigProcessor(validator, storage);
+		controller = new AuthorizationController(new Environment("test"), new DevelopmentAuthorizationConfig(), Collections.emptyList(), storage);
+		controller.init();
+		controller.start();
+	}
+	
+	@BeforeEach
+	public void setupTest(){
+
+		form = new ExportForm();
+		AbsoluteMode mode = new AbsoluteMode();
+		form.setTimeMode(mode);
+		form.setQueryGroup(new ManagedExecutionId(datasetId, UUID.randomUUID()));
+		mode.setForm(form);
+		mode.setFeatures(List.of(new CQConcept()));
+	}
+
+	@AfterEach
+	public void cleanupTest() {
+		storage.clear();
+	}
+	
+	@Test
+	public void addConfigWithoutTranslation() {
+		User user = new User("test","test");
+		storage.addUser(user);
+		user.addPermission(storage, DatasetPermission.onInstance(Ability.READ, datasetId));
+		
+		ObjectMapper mapper = FormConfigProcessor.getMAPPER();
+		FormConfigAPI formConfig = FormConfigAPI.builder()
+			.formType(form.getFormType())
+			.values(mapper.valueToTree(form))
+			.build();
+		FormConfigId formId = processor.addConfig(user, datasetId, formConfig);
+		
+		assertThat(storage.getAllFormConfigs()).containsExactly(FormConfigAPI.intern(formConfig, user.getId(), dataset.getId()));
```

Comment:
da sind noch andere sachen mit drin?

Reply:
mmhm, das sollte eigentlich schon mit drinne gewesen sein. Ich schaue es mir an

Reply:
@awildturtok das ist in Ordnung, den Storage mock braucht man ja jetzt nicht mehr